pymc_marketing/mmm/base.py

```diff
@@ -472,8 +474,136 @@ def compute_channel_contribution_original_scale(self) -> DataArray:
+        ax.legend()
+
+    def optimize_channel_budget_for_maximum_contribution(
+        self,
+        total_budget: int,
+        *,
+        parameters: Dict[str, Tuple[float, float]],
+        budget_bounds: Optional[Dict[str, Tuple[float, float]]],
+    ) -> pd.DataFrame:
+        """
+        Optimize the allocation of a given total budget across multiple channels to maximize the expected contribution.
+
+        The optimization is based on the Michaelis-Menten equation, where each channel's contribution
+        follows a saturating function of its allocated budget. The function seeks the budget allocation
+        that maximizes the total expected contribution across all channels.
+
+        Parameters
+        ----------
+        total_budget : int, requiere
+            The total budget to be distributed across channels.
+        parameters : dict, requiere
+            A dictionary where keys are channel names and values are tuples (L, k) representing the
+            Michaelis-Menten parameters for each channel.
+        budget_bounds : dict, optional
+            An optional dictionary defining the minimum and maximum budget for each channel.
+            If not provided, the budget for each channel is constrained between 0 and its L value.
+
+        Returns
+        -------
+        DataFrame
+            A pandas DataFrame containing the allocated budget and contribution information.
+
+        Raises
+        ------
+        ValueError
+            If any of the required parameters are not provided or have an incorrect type.
+        """
+        if not isinstance(budget_bounds, dict):
+            raise ValueError("The 'budget_bounds' parameter must be a dictionary.")
+
+        if not isinstance(total_budget, (int, float)):
+            raise ValueError(
+                "The 'total_budget' parameter must be an integer or float."
+            )
+
+        if not parameters:
+            raise ValueError(
+                "The 'parameters' argument (keyword-only) must be provided and non-empty."
+            )
+
+        return budget_allocator(
+            total_budget=total_budget,
+            channels=list(self.channel_columns),
+            parameters=parameters,
+            budget_ranges=budget_bounds,
+        )
+
+    def compute_channel_estimate_points_original_scale(self) -> Dict:
+        """
+        Estimate optimal and plateau points for each channel.
+
+        Returns
+        -------
+        Dict
+            A DataFrame with the estimated points.
+        """
+        channel_contributions = self.compute_channel_contribution_original_scale().mean(
+            ["chain", "draw"]
+        )
+
+        return {
+            channel: estimate_menten_parameters(channel, self.X, channel_contributions)
+            for channel in self.channel_columns
+        }
+
+    def plot_direct_contribution_curves(
+        self, show_estimations: bool = False, x_stop=None
```

Comment:
can re rename the parameter `show_estimations` to `show_michaelis_menten_fit`

Reply:
By default we should keep the original range. In the example notebook when I run 

```python
fig = mmm.plot_direct_contribution_curves(show_estimations=True)
[ax.set(xlabel="x") for ax in fig.axes]
```
I get:

![image](https://github.com/pymc-labs/pymc-marketing/assets/22996444/5fe1d778-48e1-46c7-bdf6-22e929515e02)

Reply:
I would rename `x_stop` to `xlim_max`

Reply:
Otherwise, using `x_stop = 1.5` I get 
![image](https://github.com/pymc-labs/pymc-marketing/assets/22996444/ce33aff4-8f56-427e-83e7-de7af87dc4a2)